float64 x1_lb
float64 x1_ub
float64 x2_lb
float64 x2_ub
float64 x3_lb
float64 x3_ub
